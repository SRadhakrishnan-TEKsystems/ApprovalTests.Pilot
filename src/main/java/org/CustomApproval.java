package org;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.TestInfo;
import org.junit.jupiter.api.TestInstance;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

@TestInstance(TestInstance.Lifecycle.PER_METHOD)
public class CustomApproval {
    private String className;
    private String testName;
    private String approvedFileName;
    private String path;

    @BeforeAll
    static void initialize() {
    }

    @BeforeEach
    public void initializeTest(TestInfo testInfo) {
        this.className = this.getClass().getSimpleName();
        this.testName = testInfo.getDisplayName().replace("()", "");
        this.approvedFileName = String.format("%s.%s.approved.json", className, testName);
        String partialPath = this.getClass().getPackage().getName().replaceAll("\\.", "/");
        this.path = String.format("src/test/java/%s/%s", partialPath, approvedFileName);
    }

    protected void approve(Object existing, Object actual) {
        createGoldenMaster(existing);
        JsonApprovals.verifyAsJson(actual, GsonBuilder::serializeNulls);
    }

    private void createGoldenMaster(Object existing) {
        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
        String goldenMaster = gson.toJson(existing);
        try {
            Files.write(Paths.get(path), goldenMaster.getBytes());
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
