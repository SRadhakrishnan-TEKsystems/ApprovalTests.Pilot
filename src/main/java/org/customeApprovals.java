package org;

public class customeApprovals {
}
