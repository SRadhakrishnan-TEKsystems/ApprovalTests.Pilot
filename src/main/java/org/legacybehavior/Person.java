package org.legacybehavior;

public class Person {
    private final String name;
    private final String superHeroName;
    private final String universe;

    private Person(String name, String superHeroName, String universe) {
        this.name = name;
        this.superHeroName = superHeroName;
        this.universe = universe;
    }

    public static Person getInstance(String name, String superHeroName, String universe) {
        return new Person(name, superHeroName, universe);
    }
}
