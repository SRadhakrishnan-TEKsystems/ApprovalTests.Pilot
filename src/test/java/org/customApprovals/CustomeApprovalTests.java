package org.customApprovals;

import org.CustomApproval;
import org.junit.jupiter.api.Test;
import org.legacybehavior.Person;

public class CustomeApprovalTests extends CustomApproval {
    @Test
    public void smoke() {
        Person actual = Person.getInstance("Peter Parker","Spider Man","MCU");
        Person existing = Person.getInstance("Peter Parker","Spider Man","Marvel");;
        approve(existing,actual);
    }
}
