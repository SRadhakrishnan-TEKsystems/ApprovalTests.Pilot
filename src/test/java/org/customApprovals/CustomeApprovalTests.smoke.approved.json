{
  "name": "Peter Parker",
  "superHeroName": "Spider Man",
  "universe": "Marvel"
}